src/eos-prepare-usb-update.c

```diff
@@ -418,6 +421,40 @@ run_pull_task_func (GTask *task,
       return;
     }
 
+  /* Pull the summary as well. */
+  if (!ostree_repo_remote_fetch_summary (repo,
+                                         pull_data->refspec->remote,
+                                         &summary,
+                                         &summary_sig,
+                                         cancellable,
+                                         &local_error))
+    {
+      g_task_return_error (task, g_steal_pointer (&local_error));
+      return;
+    }
+
+  repo_path = g_file_get_path (ostree_repo_get_path (repo));
+  summary_path = g_build_filename (repo_path, "summary", NULL);
+  summary_sig_path = g_build_filename (repo_path, "summary.sig", NULL);
+
+  if (!g_file_set_contents (summary_path,
+                            g_bytes_get_data (summary, NULL),
+                            g_bytes_get_size (summary),
+                            &local_error))
+    {
+      g_task_return_error (task, g_steal_pointer (&local_error));
+      return;
+    }
```

Comment:
However, if the repo has the eos-summary and eos-summary.sig files, those should probably be preferred and put at summary and summary.sig locations.